src/cdk/testing/selenium-webdriver/selenium-web-driver-element.ts

```diff
@@ -275,9 +275,26 @@ export class SeleniumWebDriverElement implements TestElement {
  * pure function, because it gets stringified by WebDriver and is executed inside the browser.
  */
 function dispatchEvent(name: string, element: Element, data?: Record<string, EventData>) {
-  const event = document.createEvent('Event');
-  event.initEvent(name);
   // tslint:disable-next-line:ban Have to use `Object.assign` to preserve the original object.
-  Object.assign(event, data || {});
+  const richEventData = Object.assign({view: window}, data || {});
+  let event: Event;
+  if (name.match(/^dblclick|click|mouse(over|enter|down|move|up|out|leave)$/)) {
```

Comment:
Not just for custom events, also to give people a way to trigger events we haven't explicitly added support for, so I think this might be a good change to make.

Reply:
It would seem unexpected to me if webdriver/protractor consumers end up using the `dispatchEvent` function without realizing that there are alternatives (like `.click`) that represent the actual e2e interaction (and not a simulated one that is using fake events; non-trusted ones).

Having individual methods like `.click/.mouseDown` will solve this as we can use the native interaction for the e2e environments, and the fake ones when needed (such as in TestBed)

Reply:
It would seem unexpected to me if webdriver/protractor consumers end up using the `dispatchEvent` function without realizing that there are alternatives (like `.click`) that represent the actual e2e interaction (and not a simulated one that is using fake events; non-trusted ones).

Having individual methods like `.click/.mouseDown` will solve this as we can use the native interaction for the e2e environments, and the fake ones when needed (such as in TestBed)

Reply:
Sounds good, let me refactor this change to include the methods that I need. That will be mousedown/move/up